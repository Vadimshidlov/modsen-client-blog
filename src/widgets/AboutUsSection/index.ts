export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
