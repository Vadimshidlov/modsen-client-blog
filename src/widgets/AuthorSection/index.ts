export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
