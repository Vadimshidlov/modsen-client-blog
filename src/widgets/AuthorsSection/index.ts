export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
