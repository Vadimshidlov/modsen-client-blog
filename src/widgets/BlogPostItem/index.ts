export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
