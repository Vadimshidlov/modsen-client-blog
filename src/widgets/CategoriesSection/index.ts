export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
