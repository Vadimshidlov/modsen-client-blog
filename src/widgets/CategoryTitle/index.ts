export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
