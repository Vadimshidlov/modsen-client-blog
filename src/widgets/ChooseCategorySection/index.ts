export { ChooseCategorySection } from '@/widgets/ChooseCategorySection/ui/ChooseCategorySection';
