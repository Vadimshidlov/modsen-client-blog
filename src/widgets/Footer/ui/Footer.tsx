import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import styles from '@/widgets/Footer/ui/Footer.module.scss';
import { NavigationBar } from '@/widgets/NavigationBar';
import { NAVIGATION_LINKS } from '@/shared/constants/navigationBar';
import { AppTitle } from '@/shared/ui/AppTitle';
import { SOCIALS } from '@/shared/constants/socials';
import { LettersForm } from '@/features/LettersForm';

export function Footer() {
  return (
    <footer className={styles.footer__container}>
      <div className={styles.footer__titleContainer}>
        <AppTitle />
        <NavigationBar navigationLinks={NAVIGATION_LINKS} />
      </div>
      <LettersForm />
      <div className={styles.footer__contactsContainer}>
        <div className={styles.footer__addressContainer}>
          <span className={styles.footer__address}>Finstreet 118 2561 Fintown</span>
          <span className={styles.footer__address}>Hello@finsweet.com 020 7993 2905</span>
        </div>
        <ul className={styles.footer__social}>
          {SOCIALS.map(({ image, name, href }) => (
            <li key={name}>
              <Link href={href}>
                <Image src={image} alt="social-pic" />
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </footer>
  );
}
