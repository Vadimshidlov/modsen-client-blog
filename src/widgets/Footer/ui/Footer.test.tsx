import { render } from '@testing-library/react';
import { Footer } from '@/widgets/Footer';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('ChooseCategorySection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
