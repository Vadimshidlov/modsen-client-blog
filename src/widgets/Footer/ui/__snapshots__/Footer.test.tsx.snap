// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCategorySection tests should matches snapshot 1`] = `
<div>
  <footer
    class="footer__container"
  >
    <div
      class="footer__titleContainer"
    >
      <h1
        class="app__title"
      >
        Mocked appTitle
      </h1>
      <ul
        class="navigation"
      >
        <li
          class="navigation__item"
        >
          <a
            data-testid="home-header-link"
            href="/"
          >
            Mocked Home
          </a>
        </li>
        <li
          class="navigation__item"
        >
          <a
            data-testid="blog-header-link"
            href="/blog"
          >
            Mocked Blog
          </a>
        </li>
        <li
          class="navigation__item"
        >
          <a
            data-testid="about us-header-link"
            href="/about"
          >
            Mocked About Us
          </a>
        </li>
        <li
          class="navigation__item"
        >
          <a
            data-testid="contact us-header-link"
            href="/"
          >
            Mocked Contact us
          </a>
        </li>
        <li
          class="navigation__item"
        >
          <a
            data-testid="privacy policy-header-link"
            href="/privacy"
          >
            Mocked Privacy Policy
          </a>
        </li>
      </ul>
    </div>
    <div
      class="form__container"
    >
      <h3
        class="form__title"
      >
        Mocked title
      </h3>
      <form
        action="#"
        class="form"
      >
        <label
          class="formLabel"
          for="email"
        >
          <input
            autocomplete="off"
            class="formInput"
            data-testid="letters-form-imput"
            id="email"
            name="email"
            placeholder="Mocked inputPlaceholder"
            type="text"
            value=""
          />
        </label>
        <button
          class="button fullWidth"
          type="submit"
        >
          Mocked buttonText
        </button>
      </form>
    </div>
    <div
      class="footer__contactsContainer"
    >
      <div
        class="footer__addressContainer"
      >
        <span
          class="footer__address"
        >
          Finstreet 118 2561 Fintown
        </span>
        <span
          class="footer__address"
        >
          Hello@finsweet.com 020 7993 2905
        </span>
      </div>
      <ul
        class="footer__social"
      >
        <li>
          <a
            href="https://www.facebook.com"
          >
            <img
              alt="social-pic"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.facebook.com"
          >
            <img
              alt="social-pic"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.facebook.com"
          >
            <img
              alt="social-pic"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.facebook.com"
          >
            <img
              alt="social-pic"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
